src/openms/source/ANALYSIS/ID/PeptideIndexing.cpp

```diff
@@ -51,20 +51,22 @@ using namespace std;
     defaults_.setValue("missing_decoy_action", "error", "Action to take if NO peptide was assigned to a decoy protein (which indicates wrong database or decoy string): 'error' (exit with error, no output), 'warn' (exit with success, warning message), 'silent' (no action is taken, not even a warning)");
     defaults_.setValidStrings("missing_decoy_action", ListUtils::create<String>("error,warn,silent"));
 
-    defaults_.setValue("enzyme:name", "Trypsin", "Enzyme which determines valid cleavage sites - e.g. trypsin cleaves after lysine (K) or arginine (R), but not before proline (P).");
+    defaults_.setValue("enzyme:name", "", "Enzyme which determines valid cleavage sites - e.g. trypsin cleaves after lysine (K) or arginine (R), but not before proline (P). Default: deduce from input");
 
-    StringList enzymes;
+    StringList enzymes{};
     ProteaseDB::getInstance()->getAllNames(enzymes);
+    enzymes.emplace_back("");
     defaults_.setValidStrings("enzyme:name", enzymes);
 
-    defaults_.setValue("enzyme:specificity", EnzymaticDigestion::NamesOfSpecificity[0], "Specificity of the enzyme."
-                                                                                               "\n  '" + EnzymaticDigestion::NamesOfSpecificity[0] + "': both internal cleavage sites must match."
-                                                                                                                                                     "\n  '" + EnzymaticDigestion::NamesOfSpecificity[1] + "': one of two internal cleavage sites must match."
-                                                                                                                                                                                                           "\n  '" + EnzymaticDigestion::NamesOfSpecificity[2] + "': allow all peptide hits no matter their context. Therefore, the enzyme chosen does not play a role here");
+    defaults_.setValue("enzyme:specificity", "", "Specificity of the enzyme. Default: deduce from input."
+      "\n  '" + EnzymaticDigestion::NamesOfSpecificity[2] + "': both internal cleavage sites must match."
```

Comment:
hmm on the left we have [0] und here we have[2] could we use the enum as index directly?